rustler_codegen/src/nif.rs

```diff
@@ -19,6 +19,15 @@ pub fn transcoder_decorator(args: syn::AttributeArgs, fun: syn::ItemFn) -> Token
         .map(|ref n| syn::Ident::new(n, Span::call_site()))
         .unwrap_or_else(|| name.clone());
 
+    if fun.sig.generics.type_params().next().is_some() {
```

Comment:
Can you give a small example where this panic would be triggered?

Reply:
```rust
#[rustler::nif]
pub fn generic_echo<T>(t: T) -> T {
    t
}
```
This is a bit of a silly example, so here is a more relevant one:

```rust
// Ok, derive now works for polymorphic types
#[derive(NifTuple)]
pub struct GenericTuple<T> {
    t1: T,
    t2: T,
}

// This is fine, generic type is fully applied
#[rustler::nif]
pub fn generic_tuple_echo_usize(generic_tuple: GenericTuple<usize>) -> GenericTuple<usize> {
    generic_tuple
}

// This could type check, but rustler cannot work with it
#[rustler::nif]
pub fn generic_tuple_echo<T>(generic_tuple: GenericTuple<T>) -> GenericTuple<T> {
    generic_tuple
}
```

Reply:
Thanks!